execute/group_transformation_test.go

```diff
@@ -0,0 +1,262 @@
+package execute_test
+
+import (
+	"testing"
+
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/execute/table/static"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/mock"
+	"github.com/influxdata/flux/values"
+)
+
+func TestGroupTransformation_ProcessChunk(t *testing.T) {
+	// Ensure we allocate and free all memory correctly.
+	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
+	defer mem.AssertSize(t, 0)
+
+	// Generate one table chunk using static.Table.
+	// This will only produce one column reader, so we are
+	// extracting that value from the nested iterators.
+	want := static.Table{
+		static.Times("_time", 0, 10, 20),
+		static.Floats("_value", 1, 2, 3),
+	}
+
+	isProcessed := false
+	tr, _, err := execute.NewGroupTransformation(
+		executetest.RandomDatasetID(),
+		&mock.GroupTransformation{
+			ProcessFn: func(chunk table.Chunk, d *execute.TransportDataset, _ memory.Allocator) error {
+				// Memory should be allocated and should not have been improperly freed.
+				// This accounts for 64 bytes (data) + 64 bytes (null bitmap) for each column
+				// of which there are two. 64 bytes is the minimum that arrow will allocate
+				// for a particular data buffer.
+				mem.AssertSize(t, 256)
+
+				// Compare the buffer contents to the table we wanted.
+				// Because we should have produced only one table chunk,
+				// we are comparing the entirety of the chunk to the entirety
+				// of the wanted output.
+				buffer := chunk.Buffer()
+				got := table.Iterator{
+					table.FromBuffer(&buffer),
+				}
+
+				if diff := table.Diff(want, got); diff != "" {
+					t.Errorf("unexpected diff -want/+got:\n%s", diff)
+				}
+				isProcessed = true
+				return nil
+			},
+		},
+		mem,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// We can use a TransportDataset as a mock source
+	// to send messages to the transformation we are testing.
+	source := execute.NewTransportDataset(executetest.RandomDatasetID(), mem)
+	source.AddTransformation(tr)
+
+	tbl := want.Table(mem)
+	if err := tbl.Do(func(cr flux.ColReader) error {
+		chunk := table.ChunkFromReader(cr)
+		chunk.Retain()
+		return source.Process(chunk)
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	if !isProcessed {
+		t.Error("message was never processed")
+	}
+}
+
+func TestGroupTransformation_FlushKey(t *testing.T) {
```

Comment:
This test case looks a bit confused. There's no mention of group transformation in the test itself. I think it would be good to use group transformation and confirm that `FlushKey` does not get forwarded.

I just took a look in the code and it appears the `NarrowTransformation` version of this is similarly confused. So I'll need to go back and update that test case.